hecuba/storageobj.py

```diff
@@ -350,43 +348,56 @@ def make_persistent(self, name):
             log.error("Unable to execute %s", query_simple)
             raise ir
 
+        #Iterate over the objects the user has requested to be persistent
+        # retrieve them from memory and make them persistent
         for obj_name, obj_info in self._persistent_props.items():
-            if hasattr(self, obj_name):
-                pd = getattr(self, obj_name)
+            try:
+                pd = object.__getattribute__(self, obj_name)
                 if obj_info['type'] not in IStorage._basic_types:
                     sd_name = self._ksp + "." + self._table + "_" + obj_name
                     pd.make_persistent(sd_name)
-                setattr(self, obj_name, pd)  # super(StorageObj, self).__setattr__(obj_name, pd) why?
-
+                #self is persistent so setting the attribute will store the data and create the appropiate binding
+                setattr(self, obj_name, pd)
+            except AttributeError:
+                #Attribute unset, no action needed
+                pass
         self._store_meta(self._build_args)
 
     def stop_persistent(self):
         """
             The StorageObj stops being persistent, but keeps the information already stored in Cassandra
         """
+        for obj_name, obj_info in self._persistent_props.items():
+            if obj_info['type'] not in IStorage._basic_types:
+                try:
+                    pd = object.__getattribute__(self, obj_name)
+                    pd.stop_persistent()
+                except AttributeError:
+                    # Attribute unset, no action needed
+                    pass
         log.debug("STOP PERSISTENT")
-        for persistent_dict in self._persistent_dicts:
-            persistent_dict.stop_persistent()
+        self._is_persistent = False
 
     def delete_persistent(self):
         """
             Deletes the Cassandra table where the persistent StorageObj stores data
         """
-        self._is_persistent = False
-
-        if hasattr(self, '_persistent_dicts'):
-            for pers_dict in self._persistent_dicts:
-                pers_dict.delete_persistent()
 
-        if hasattr(self, '_storage_objs'):
-            for so in self._storage_objs:
-                so.delete_persistent()
+        for (attr_name,attr_info) in self._persistent_props.iteritems():
+            if attr_info['type'] not in IStorage._basic_types:
+                try:
+                    pers_obj = getattr(self,attr_name)
+                    pers_obj.delete_persistent()
+                except AttributeError as ex:
+                    pass
 
         query = "TRUNCATE TABLE %s.%s;" % (self._ksp, self._table)
         log.debug("DELETE PERSISTENT: %s", query)
         config.session.execute(query)
 
-    def __getattr__(self, attribute):
+        self._is_persistent = False
+
+    def __getattribute__(self, attribute):
```

Comment:
you CANNOT override the __getattribute__ ! If you do you won't be able to access to any class method.

Reply:
check the __getattr__